linux_os/guide/services/sssd/sssd_enable_smartcards/ansible/shared.yml

```diff
@@ -33,3 +33,179 @@
+        dest: "/etc/authselect/{{ authselect_custom_profile }}/system-auth"
+        regexp: '^(auth.*sufficient.*pam_sss\.so)((?!try_cert_auth).)*$'
+        replace: '\g<1> try_cert_auth \g<2>'
+      when:
+        - result_authselect_profile is not skipped
+        - result_pam_try_cert_auth_present_system_auth.found == 0
+
+    - name: Ensure a backup of current authselect profile before selecting the custom profile
+      ansible.builtin.command:
+        cmd: authselect apply-changes -b --backup=before-pwhistory-hardening.backup
+      register: result_authselect_backup
+      when:
+        - result_authselect_check_cmd is success
+        - result_authselect_profile is not skipped
+        - authselect_current_profile is not match("custom/")
+        - authselect_custom_profile is not match(authselect_current_profile)
+
+    - name: Ensure the custom profile is selected
+      ansible.builtin.command:
+        cmd: authselect select {{ authselect_custom_profile }} --force
+      register: result_pam_authselect_select_profile
+      when:
+        - result_authselect_check_cmd is success
+        - result_authselect_profile is not skipped
+        - authselect_current_profile is not match("custom/")
+        - authselect_custom_profile is not match(authselect_current_profile)
+
+    - name: Restore the authselect features in the custom profile
+      ansible.builtin.command:
+        cmd: authselect enable-feature {{ item }}
+      loop: "{{ result_authselect_features.stdout_lines }}"
+      when:
+        - result_authselect_profile is not skipped
+        - result_authselect_features is not skipped
+        - result_pam_authselect_select_profile is not skipped
+
+    - name: Ensure the custom profile changes are applied
+      ansible.builtin.command:
+        cmd: authselect apply-changes -b --backup=after-pwhistory-hardening.backup
+      when:
+        - result_authselect_check_cmd is success
+        - result_authselect_profile is not skipped
+  when:
+    - (result_pam_try_cert_auth_present_smartcard_auth.found == 0) or (result_pam_try_cert_auth_present_system_auth.found == 0)
+    - result_authselect_present.stat.exists
+
+# For systems without authselect
+- name: "Remediation where authselect tool is not present and PAM files are directly edited"
+  block:
+    - name: Ensure the try_cert_auth option is in smartcard-auth
+      ansible.builtin.replace:
+        dest: "/etc/pam.d/smartcard-auth"
+        regexp: '^(auth.*sufficient.*pam_sss\.so)((?!try_cert_auth).)*$'
+        replace: '\g<1> try_cert_auth \g<2>'
+      when:
+        - result_pam_try_cert_auth_present_smartcard_auth.found == 0
+
+    - name: Ensure the try_cert_auth option is in system-auth
+      ansible.builtin.replace:
+        dest: "/etc/pam.d/system-auth"
+        regexp: '^(auth.*sufficient.*pam_sss\.so)((?!try_cert_auth).)*$'
+        replace: '\g<1> try_cert_auth \g<2>'
+      when:
+          - result_pam_try_cert_auth_present_system_auth.found == 0
```

Comment:
If I understand it well, the OVAL check requires that the setting is present in both smartcard-auth and system-auth, there is an AND between the criteria. So then the Ansible remdiation should create both of them.